### DISCUSSION TITLE
Allow the exact email address to be used in the dialog to generate a username

### DISCUSSION THREAD

**Discussion by igbominadeveloper** [bitwarden/clients#6592] [Password Manager]
### Code Contribution Proposal

I usually use Bitwarden to complete signup forms and in the process, I need my email address in the username field. Now, since I can generate a username with the dialog, I usually still have to edit the username field because all the options add some characters to my email address - this feature works as it should.
An improvement I would like to add to it as an option to use my exact email address as it is.

https://community.bitwarden.com/t/autofill-the-username-field-when-adding-a-new-login/59121

**Comment by djsmith85** [comment]
Adding a link to the accompanying PR: https://github.com/bitwarden/clients/pull/6593

@bitwarden/dept-product: Contributor is looking for guidance on how to proceed with this feature, see community forum thread.

**Reply by micahblut** [reply]
Bitwarden would prefer users make use of the existing generator features to set up unique email addresses for different accounts, because this is a better security practice. We will not be accepting any contributions to enable a default email address to populate for username.

**Reply by igbominadeveloper** [reply]
@micahblut. I am curious, could you help explain what the risk is with allow the user to populate the username field with the email address signed up with on bitwarden? Thanks.

**Reply by micahblut** [reply]
@igbominadeveloper If you use the same email address across different services and one or more of those services are breached (an inevitability), it can be very easy for hackers and bad actors to cross-reference your details and link your information together.

Even worse is if you use the same password all over the place too. A hacker could then try other services (e.g. PayPal) with that same email and password combination, allowing them to potentially gain access to your account.

**Reply by igbominadeveloper** [reply]
Alright. Thanks for the explanation. I will close the PR and work on the next one #6601

**Reply by hessam-kk** [reply]
Ok, I see your perspective, but wouldn't it be better to let people choose what to do? instead of forcing them to do specific tasks? I really need integration, and I really, really want to use the same e-mail address for all my accounts. There is no option to import my default email and use it everywhere. It is quite annoying that I need to type my lengthy email address each time.

**Reply by micahblut** [reply]
@hessam-kk I understand it can be annoying to type a lengthy email address every time, but that's your choice, and Bitwarden does not stop you from making that choice. Bitwarden just don't make that choice convenient, because we want to bias users toward best practices.

**Reply by teambob** [reply]
Subaddresses can be linked by stripping the different part of the regex. Certainly having different *passwords* is considered best practice but could you provide references where using different *usernames* is considered best practice?

I was half way through making my own PR when I noticed this one.

**Reply by gfra54** [reply]
> @hessam-kk I understand it can be annoying to type a lengthy email address every time, but that's your choice, and Bitwarden does not stop you from making that choice. Bitwarden just don't make that choice convenient, because we want to bias users toward best practices.

I respectfully disagree. While using unique email addresses might be helpful in certain cases, I don't see it as a best practice. With strong passwords and especially 2FA, the added value seems minimal. For many users—like my mom, who uses the same password everywhere—introducing such measures from scratch can be overwhelming and counterproductive, making it harder to adopt essential best practices.

Moreover, relying on third-party services for email redirection creates dependency and complicates future migrations. It adds complexity and potential points of failure. Trusting 3rd party providers like DuckDuckGo or ForwardEmail (Who are those people ?) with my online identity feels like an unnecessary risk. I already trust my email provider, and Bitwarden, this is enought stress for me😅

To go back to the original topic, I think that it could be very helpull to have autocomplete based on existing usernames in my bitwarden vault in the username field when I add new credentials.

**Reply by detly** [reply]
> it can be very easy for hackers and bad actors to cross-reference your details and link your information together.

It seems like this is predicated on those bad actors being unaware of the plus-addressing trick. But I think they might have caught on to it by now?

Also:

- many services and brokers will strip such suffixes and store only the canonical address anyway
- not everyone is using GMail

**Comment by hessam-kk** [comment]
It has nothing to do with usernames; many websites only require email
addresses to log in, so I am forced to enter my email address. I will not
get an email to activate my account if I add anything to the end of my
email address.

**Reply by igbominadeveloper** [reply]
I made another PR that allows you to make using your exact email address easier by turning the username input into an autocomplete input - https://github.com/bitwarden/clients/pull/6601

Would be really useful if it gets merged
